tfs/src/main/java/hudson/plugins/tfs/listeners/JenkinsRunListener.java

```diff
@@ -54,4 +93,24 @@ private String getStartedBy(final Run run) {
         }
         return startedBy;
     }
+
+    private JSONObject createJsonFromRun(final Run run) {
+        return new JSONObject();
+    }
+
+    private String getRunConfig(final Run run) {
+        if (run instanceof WorkflowRun) {
+            List<? extends Action> actionList = ((WorkflowRun) run).getAllActions();
+            for (final Action currAction : actionList) {
+                if (currAction instanceof CauseAction) {
+                    for (final Cause currCause : ((CauseAction) currAction).getCauses()) {
+                        if (currCause instanceof TeamPushCause) {
+                            return ((TeamPushCause) currCause).getRunConfig();
+                        }
+                    }
+                }
+            }
+        }
+        return "Jenkins CI build";
```

Comment:
Can you pull this string out into a static constant somewhere?

Reply:
Good suggestion. Refactored it as a static constant in JenkinsRunListener class:
private static final String DEFAULT_RUN_CONTEXT = "Jenkins CI build";